WordPress/Classes/ViewRelated/Plugins/PluginDirectoryViewController.swift

```diff
@@ -50,27 +52,97 @@ class PluginDirectoryViewController: UITableViewController {
         handler.viewModel = viewModel.tableViewModel(presenter: self)
 
         immuHandler = handler
+
+        navigationItem.rightBarButtonItem = searchBarButton
+
+        setupSearchBar()
     }
 
     override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
-        return 0
+        return CGFloat.leastNonzeroMagnitude
     }
 
     override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
-        return 0
+        return CGFloat.leastNonzeroMagnitude
     }
 
     private func reloadTable() {
         immuHandler?.viewModel = viewModel.tableViewModel(presenter: self)
     }
 
+    @objc private func searchButtonTapped() {
+        searchController.isActive = true
+    }
+
+    private func setupSearchBar() {
+        let containerView = UIView(frame: CGRect(origin: .zero,
+                                                 size: CGSize(width: tableView.frame.width,
+                                                              height: searchController.searchBar.frame.height)))
+        containerView.addSubview(searchController.searchBar)
+        tableView.tableHeaderView = containerView
+        // for some... particlar reason, which I haven't been able to fully track down, if the searchBar is added directly
+        // as the tableHeaderView, the UITableView sort of freaks out and adds like 400pts of random padding
+        // below the content of the tableView. Wrapping it in this container fixes it ¯\_(ツ)_/¯
```

Comment:
I was looking at `PostListViewController` and I saw it also uses a wrapper view for the search bar. Maybe @frosty has more insights about this? It looks fine to me, but since his approach was slightly different there might be something we're missing.